src/main/java/gov/nasa/jpl/mbee/mdk/mms/endpoints/MMSEndpoint.java

```diff
@@ -0,0 +1,111 @@
+            e.printStackTrace();
+        }
+    }
+
+    public abstract void prepareUriPath();
+
+    public URIBuilder getEndpoint() { return uriBuilder; }
+
+    public void setParameter(String key, String value) {
+        uriBuilder.setParameter(key, value);
+    }
+
+    /**
+     * General purpose method for making http requests for JSON objects. Type of request is specified in method call.
+     *
+     * @param type       Type of request, as selected from one of the options in the inner enum.
+     * @param sendData   Data to send as an entity/body along with the request, if desired. Support for GET and DELETE
+     *                   with body is included.
+     * @return
+     * @throws IOException
+     * @throws URISyntaxException
+     */
+    public HttpRequestBase buildRequest(MMSUtils.HttpRequestType type, File sendData, ContentType contentType, Project project) throws IOException, URISyntaxException {
+        if(uriBuilder != null) {
+            // build specified request type
+            // assume that any request can have a body, and just build the appropriate one
+            URI requestDest = uriBuilder.build();
+            final HttpRequestBase request;
+            // bulk GETs are not supported in MMS, but bulk PUTs are. checking and and throwing error here in case
+            if (type == MMSUtils.HttpRequestType.GET && sendData != null) {
+                throw new IOException("GETs with body are not supported");
+            }
+            switch (type) {
+                case DELETE:
+                    request = new HttpDeleteWithBody(requestDest);
+                    break;
+                case GET:
+                default:
+                    request = new HttpGet(requestDest);
+                    break;
+                case POST:
+                    request = new HttpPost(requestDest);
+                    break;
+                case PUT:
+                    request = new HttpPut(requestDest);
+                    break;
+            }
+            request.addHeader("Authorization", "Bearer " + TicketUtils.getTicket(project));
+            request.addHeader("Content-Type", "application/json"); // is this good logic?
+            request.addHeader("charset", (contentType != null ? contentType.getCharset() : Consts.UTF_8).displayName());
+            if (sendData != null) {
+                if (contentType != null) {
+                    request.addHeader("Content-Type", contentType.getMimeType());
+                }
+                HttpEntity reqEntity = new FileEntity(sendData, contentType);
+                //reqEntity.setChunked(true);
+                ((HttpEntityEnclosingRequest) request).setEntity(reqEntity);
+            }
+            return request;
+        }
+        return null;
+    }
+
+    public HttpRequestBase buildImageRequest(File sendFile, Project project) throws URISyntaxException {
```

Comment:
Why did we have to introduce a different method for images vs. using `buildRequest`? If it was exclusively for `ExportImage`, then it's not necessary as that class is no longer in use and doesn't need to be migrated I believe.